puzzlejs/puzzle.js

```diff
@@ -1384,9 +1385,11 @@ function PuzzleEntry(p, index) {
         this.commands.querySelector(".puzzle-about-button").addEventListener("click", e => { this.aboutPopup(); });
         this.commands.querySelector(".puzzle-undo-button").addEventListener("click", e => { this.undoManager.undo(); });
         this.commands.querySelector(".puzzle-redo-button").addEventListener("click", e => { this.undoManager.redo(); });
-        // TODO shouldn't need a reload
-        this.commands.querySelector(".puzzle-reset-button").addEventListener("click", e => { this.prepareToReset(); window.location.reload(); });
-
+        // TODO shouldn't need a reload after reset
+        if (this.options["data-reset-prompt"])
+            this.commands.querySelector(".puzzle-reset-button").addEventListener("click", e => { if (confirm(this.options["data-reset-prompt"])) {this.prepareToReset(); window.location.reload();} });
```

Comment:
A few things here:

1. I think these four lines could just boil down to using this handler: `e => { var prompt = this.options["data-reset-prompt"]; if (!prompt || confirm(prompt)) { this.prepareToReset(); window.location.reload(); } }`, so that any changes to the reset logic automatically apply to both cases. That said, see item 2 below which would modify this code somewhat (but the pattern would still work fine IMO).
2. How common do you think it is that the text will change based on the puzzle? I wouldn't think this would happen often, but maybe you have ideas. Just wondering if this should be a bool with standard text (or, if we're feeling fancy, a `data-reset-prompt` bool and a `data-reset-prompt-text` string).
3. If this becomes a bool, I am 100% fine with shifting the default to "true".

Reply:
I wouldn't expect the text to change based on puzzle, but didn't want to assume that others would want the feature or my wording. Making a bool with a generic prompt is fine. What wording do you want? (I'm using "Clear all puzzle content?") Your call on whether the default is true. I'd like that, but how do you feel about changing functionality? Do you also want to include a separate prompt-override string?

Reply:
Actually, I don't think it needs 2 params, especially if you want the default to be showing the prompt. We'd just set `"data-reset-prompt": "Clear all puzzle content?"`, with override options of custom string or "" for no prompt (which would work with your code about since `!prompt` would be true). That functions just like a bool defaulting to true, but without the need to set/check multiple vars.

Reply:
> these four lines could just boil down to using this handler: `e => { var prompt = this.options["data-reset-prompt"]; if (!prompt || confirm(prompt)) { this.prepareToReset(); window.location.reload(); } }`

Changed. I'll wait to upload until we've got the prompt option design resolved.

Reply:
I think I'm now OK with the prompt string being the only option now, especially since `data-reset-prompt = ""` and `data-reset-prompt = "false"` should both succeed in turning off the prompt (since we already convert `"false"` to `false` everywhere). Thanks!